fpmsyncd/routesync.cpp

```diff
@@ -886,6 +967,62 @@ bool RouteSync::getIfName(int if_index, char *if_name, size_t name_len)
     return true;
 }
 
+/*
+ * Get next hop List
+ * @arg route_obj     route object
+ *
+ * Return concatenation of NHs: nh0 + "," + nh1 + .... + "," + nhN
+ */
+string RouteSync::getNextHopList(struct rtnl_route *route_obj)
+{
+    string result = "";
+
+    for (int i = 0; i < rtnl_route_get_nnexthops(route_obj); i++)
+    {
+        struct rtnl_nexthop *nexthop = rtnl_route_nexthop_n(route_obj, i);
+        struct nl_addr *addr = NULL;
+
+        if ((addr = rtnl_route_nh_get_newdst(nexthop)))
+        {
+            /* Next hop label is not empty */
+            char label[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, label, MAX_ADDR_SIZE);
+            result += label;
+            result += string("+");
+        }
+        else if ((addr = rtnl_route_nh_get_encap_mpls_dst(nexthop)))
+        {
+            /* Next hop encap is not empty */
+            char label[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, label, MAX_ADDR_SIZE);
+            result += label;
+            result += string("+");
+        }
+
+        if ((addr = rtnl_route_nh_get_gateway(nexthop)))
+        {
+            /* Next hop gateway is not empty */
+            char gw_ip[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, gw_ip, MAX_ADDR_SIZE);
+            result += gw_ip;
+        }
+        else if ((addr = rtnl_route_nh_get_via(nexthop)))
+        {
+            /* Next hop via is not empty */
+            char via_ip[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, via_ip, MAX_ADDR_SIZE);
+            result += via_ip;
+        }
```

Comment:
what if 'addr' is NULL? Any reason why the existing code to add "0.0.0.0" or "::" is not here.

Reply:
Done